cmd/site-replication.go

```diff
@@ -3058,3 +3272,995 @@ func (c *SiteReplicationSys) PeerEditReq(ctx context.Context, arg madmin.PeerInf
 	}
 	return nil
 }
+
+const siteHealTimeInterval = 1 * time.Minute
+
+func (c *SiteReplicationSys) startHealRoutine(ctx context.Context) {
+	healTimer := time.NewTimer(siteHealTimeInterval)
+	defer healTimer.Stop()
+
+	for {
+		select {
+		case <-healTimer.C:
+			healTimer.Reset(siteHealTimeInterval)
+			objAPI := newObjectLayerFn()
+			if objAPI == nil {
+				continue
+			}
+			c.RLock()
+			if !c.enabled {
+				continue
+			}
+			c.RUnlock()
```

Comment:
```suggestion
			c.RLock()
			enabled := c.enabled
			c.RUnlock()
			if !enabled {
				continue
			}
			
```

It can cause lock starvation otherwise

Reply:
this is a deadlock and a bug.